extensions/rich_text_components/Math/directives/oppia-noninteractive-math.component.ts

```diff
@@ -0,0 +1,124 @@
+
+export interface MathExpression {
+  'raw_latex': string;
+  'svg_filename': string;
+}
+
+@Component({
+  selector: 'oppia-noninteractive-math',
+  templateUrl: './math.component.html',
+  styleUrls: []
+})
+export class NoninteractiveMath implements OnInit {
+  @Input() mathContentWithValue: string;
+  imageContainerStyle: {
+    height: string, width: string, 'vertical-align': string};
+  imageUrl: string;
+
+  constructor(
+    private assetsBackendApiService: AssetsBackendApiService,
+    private contextService: ContextService,
+    private htmlEscaperService: HtmlEscaperService,
+    private imageLocalStorageService: ImageLocalStorageService,
+    private imagePreloaderService: ImagePreloaderService
+  ) {}
+
+  ngOnInit(): void {
+    const mathExpressionContent = this.htmlEscaperService.escapedJsonToObj(
+      this.mathContentWithValue) as MathExpression;
+    if (mathExpressionContent.hasOwnProperty('raw_latex')) {
+      const svgFilename = mathExpressionContent.svg_filename;
+      const dimensions = this.imagePreloaderService.getDimensionsOfMathSvg(
+        svgFilename);
+      this.imageContainerStyle = {
+        height: dimensions.height + 'ex',
+        width: dimensions.width + 'ex',
+        'vertical-align': '-' + dimensions.verticalPadding + 'ex'
+      };
+    }
+    // If viewing a concept card in the exploration player, don't use the
+    // preloader service. Since, in that service, the image file names are
+    // extracted from the state contents to be preloaded, but when
+    // viewing a concept, the names are not available until the link is
+    // clicked, at which point an API call is done to get the skill
+    // details. So, the image file name will not be available to the
+    // preloader service beforehand.
+    if (
+      this.imagePreloaderService.inExplorationPlayer() &&
+      !(this.contextService.getEntityType() === AppConstants.ENTITY_TYPE.SKILL)
+    ) {
+      this.imagePreloaderService.getImageUrlAsync(
+        mathExpressionContent.svg_filename
+      ).then(
+        objectUrl => this.imageUrl = objectUrl
+      );
+    } else {
+      // This is the case when user is not in the exploration player. We
+      // don't pre-load the images in this case. So we directly assign
+      // the url to the imageUrl.
+      try {
+        // For IMAGE_SAVE_DESTINATION_LOCAL_STORAGE mode we first try to
+        // fetch images through the storage and if it doesn't exist we try
+        // fetching it through server.
+        // This is required for the translation suggestion as there can be
+        // target entity's images in the translatable content which needs
+        // to be fetched from the server.
+        if (
+          this.contextService.getImageSaveDestination() ===
+          AppConstants.IMAGE_SAVE_DESTINATION_LOCAL_STORAGE && (
+            this.imageLocalStorageService.isInStorage(
+              mathExpressionContent.svg_filename))) {
```

Comment:
I think it's fine. It is a simple and condition. Due the how long our names are, it might come off as unreadable but upon reading the condition, it looks fine (having a separate would be definitely better). Its just that I am a bit reluctant on refactors that are not very important on migration PRs.

Reply:
Okay, seems fine.